Description: apply gstreamer pkg-config definitions
 To ensure gstreamer builds correctly, pick up its cflags in FindGstreamer
 and apply them to the build as definitions
Author: José Manuel Santamaría <panfaust@gmail.com>
Origin: vendor, https://lists.ubuntu.com/archives/kubuntu-devel/2015-August/009819.html
Forwarded: https://lists.ubuntu.com/archives/kubuntu-devel/2015-August/009819.html
Reviewed-by: Harald Sitter <sitter@kde.org>
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/cmake/modules/FindGStreamer.cmake
+++ b/cmake/modules/FindGStreamer.cmake
@@ -36,6 +36,7 @@ if (PKG_CONFIG_FOUND)
     exec_program(${PKG_CONFIG_EXECUTABLE}
                  ARGS --variable pluginsdir gstreamer-${GSTREAMER_ABI_VERSION}
                  OUTPUT_VARIABLE PKG_GSTREAMER_PLUGIN_DIR)
+    set(GSTREAMER_DEFINITIONS ${PKG_GSTREAMER_CFLAGS})
 endif()
 
 find_library(GSTREAMER_LIBRARY
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -106,6 +106,7 @@ find_package(GObject)
 macro_log_feature(GOBJECT_FOUND "GObject" "Required to build QtGLib" "http://www.gtk.org/" TRUE)
 
 set(CMAKE_REQUIRED_INCLUDES ${QTGSTREAMER_INCLUDES})
+add_definitions(${GSTREAMER_DEFINITIONS})
 include(CheckCXXSourceCompiles)
 check_cxx_source_compiles("
 #include <QtCore/QtGlobal>
